providers/parseable/src/lib.rs

```diff
@@ -0,0 +1,178 @@
+}
+
+#[derive(QuerySchema, Deserialize, Serialize, Debug, Clone)]
+struct Query {
+    #[pdk(label = "query", placeholder = "select * from stream")]
+    pub query: String,
+
+    #[pdk(label = "Specify a time range")]
+    pub time_range: DateTimeRange,
+}
+
+pdk_query_types! {
+    EVENTS_QUERY_TYPE => {
+        label: "Parseable: Run a query",
+        handler: query_handler(Query, Config).await,
+        supported_mime_types: [EVENTS_MIME_TYPE],
+    },
+    STATUS_QUERY_TYPE => {
+        handler: check_status(ProviderRequest).await,
+        supported_mime_types: [STATUS_MIME_TYPE]
+    }
+}
+
+#[pdk_export]
+fn create_cells(query_type: String, _response: Blob) -> Result<Vec<Cell>> {
+    match query_type.as_str() {
+        EVENTS_QUERY_TYPE => create_log_cell(),
+        _ => Err(Error::UnsupportedRequest),
+    }
+}
+
+pub fn create_log_cell() -> Result<Vec<Cell>> {
+    let logs_cell = Cell::Log(
+        LogCell::builder()
+            .id("query-results".to_string())
+            .data_links(vec![format!("cell-data:{EVENTS_MIME_TYPE},self")])
+            .hide_similar_values(false)
+            .build(),
+    );
+    Ok(vec![logs_cell])
+}
+
+async fn query_handler(query: Query, config: Config) -> Result<Blob> {
+    let events = run_query(&query, &config).await?;
+    Events(events).to_blob()
+}
+
+async fn check_status(request: ProviderRequest) -> Result<Blob> {
+    let config = Config::parse(request.config)?;
+    let url = get_url(LIVENESS_API, &config)?;
+    let response = make_http_request(HttpRequest::get(url)).await?;
+    if response.status_code == 200 {
+        ProviderStatus::builder()
+            .status(Ok(()))
+            .version(COMMIT_HASH.to_owned())
+            .built_at(BUILD_TIMESTAMP.to_owned())
+            .build()
+            .to_blob()
+    } else {
+        Err(Error::Http {
+            error: HttpRequestError::Offline,
+        })
+    }
+}
+
+async fn run_query(query: &Query, config: &Config) -> Result<Vec<ProviderEvent>> {
+    let query = json!({
+        "query": query.query,
+        "startTime": query.time_range.from.format(&well_known::Rfc3339).unwrap(),
+        "endTime": query.time_range.to.format(&well_known::Rfc3339).unwrap()
+    });
+
+    let url = get_url(QUERY_API, config)?;
+    let mut headers = BTreeMap::new();
+    headers.insert(
+        "Authorization".to_string(),
+        basic_auth_header(&config.username, &config.password),
+    );
+    headers.insert("Content-Type".to_string(), "application/json".to_string());
+
+    let request = HttpRequest::post(url, query.to_string()).with_headers(headers);
+    let response = make_http_request(request).await?;
+
+    let body: Value = serde_json::from_slice(&response.body)?;
+    let Value::Array(arr) = body  else {
+        return Err(Error::Other { message: String::from_utf8_lossy(&response.body).to_string() })
+    };
+
+    let mut rows = Vec::new();
+    for value in arr {
+        rows.push(parse_row(value))
+    }
```

Comment:
I think this can be simplified a little:

```suggestion
    let rows: Vec<_> = arr.into_iter().map(parse_row).collect();
```

Reply:
Note that I suggested below that `parse_row()` should return an error type instead. You can still use the approach suggested above, since `collect()` can collect errors too. Maybe you can suffice by just adding a question mark at the end, but if needs more type annotations, it may need to become:

```suggestion
    let rows = arr.into_iter().map(parse_row).collect::<Result<Vec<_>>()?;
```

Reply:
If parse_row is fallible then this would require `try_collect()` instead